.github/workflows/premerge_amazonlinux2_cabal-build-all.sh_-l.yml

```diff
@@ -0,0 +1,19 @@
+name: Pre-merge AmazonLinux2 Cabal Build w/ system libsodium
+
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - scripts/cnode-helper-scripts/cabal-build-all.sh
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: container login
+      run: |
+        docker login -u ${{ secrets.DOCKER_HUB_USERNAME }} -p ${{ secrets.DOCKER_HUB_PASSWORD }}
+    - name: Testing prereqs.sh -l and cabal-build-all.sh -l
+      run: |
+        docker build . --file files/tests/pre-merge/amazonlinux2-cabal_-l.containerfile --compress --tag chaincrucial/pre-merge:amazonlinux2_cabal_-l
```

Comment:
Can we make this a second job within same action file (one with system libsodium based cabal-build-all.sh and second job with IO fork)? Helps reduce workflow quota from github, tho we prolly wouldnt have frequent changes to these files, but atleast combining tests for libsodium directly divides the workflows by 2 (other alternate would be using all OS against a specific test within same action file), either of those should be alright

Reply:
Sure. I had them split to increase observability by running everything in parallel, but I could easily see an active repository exceeding the pool of free Action minutes per month. I'l streamline it per OS. Also in another week CentOS 8 could be dropped due to EOL. The RockyLinux8 tests have produced identical results to all CentOS8 build testing.

The AmazonLinux2 image appears to be based on an fc35 kernel but without dnf. The `--allowerasing` "yum" subcommand is not really part of yum, but the `alias yum='dnf'` from el8 distro's hides that. This blocks the package installation on AL2. I presume the same goes for any el7 based distro currently. I don't use el7 anymore, or AL2 for anything. If the quantity of users does not warrant supporting it with prereqs.sh, that test could also be dropped.

Reply:
Actually I guess its just 20 concurrent jobs in the queue since its a public repository, and the 2000 minutes per month limit would be only if the repo was private?

In any case I will work to reduce the concurrency and the OS's that you are not concerned with testing.

Reply:
I agree on CentOS not being essential (especially since you cover Rocky), about AL2 - yes, there are a few consumers as per issues seen previously, so would be good to retain.
Thanks

Reply:
OK. I'll submit a separate PR to add support for AL2 in prereqs.sh.